### ISSUE TITLE
The use of CuDNN CTC

### ISSUE THREAD

**Issue by kan-bayashi** [espnet/espnet#433]
Pytorch supports cndnn CTC in version 1.0.
https://pytorch.org/docs/master/nn.html?highlight=ctc#torch.nn.CTCLoss

We will consider the use of it.

**Comment by sw005320** [comment]
chainer may also support it soon by @aonotas

**Comment by fanlu** [comment]
Is there any performance and speed comparison between cudnn ctc and warp ctc? thanks.

**Comment by siddalmia** [comment]
You can see the speed comparisons here - http://danielgalvez.me/jekyll/update/2017/12/29/benchmarking-ctc-implementations.html

Though these would be the issues in using CuDNN CTC -
1. Maximum number of labels in any mini-batch member = 256 -- ( this is the same with tensorflow ) 
2. You must provide probabilities which have been calculated by softmax -- (this is slow and can have numerical instability)
3. The gradient values are with respect to pre-softmax activations and not with respect to the probabilities -- (just weird)

**Comment by aonotas** [comment]
I'm trying to merge cuDNN CTC in Chainer.

https://github.com/chainer/chainer/pull/5650
https://github.com/cupy/cupy/pull/1769

I will let you know when these PRs merged.
